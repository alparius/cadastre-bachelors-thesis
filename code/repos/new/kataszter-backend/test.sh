#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
