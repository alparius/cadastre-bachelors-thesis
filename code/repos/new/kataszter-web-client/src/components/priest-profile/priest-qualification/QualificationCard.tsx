import React from "react";
import { Button, Card, Header, List } from "semantic-ui-react";

import IQualification from "../../../data/priest/Qualification";
import "../../../static/priestProfile.css";
import CardRow from "../CardRow";
import CardRowForDate from "../CardRowForDate";

interface IQualificationCardProps {
    qualification: IQualification;
    index: number;
    editMode: boolean;
    handleDelete: (index: number) => void;
    handleChangeOnQualification: (labelName: string, value: any, index: number) => void;
}

const QualificationCard: React.FC<IQualificationCardProps> = (props: IQualificationCardProps) => {
    const { qualification, index, editMode, handleDelete, handleChangeOnQualification } = props;

    const handleDeletePressed = () => {
        handleDelete(index);
    };

    return (
        <Card fluid>
            <Card.Content>
                <Card.Content>
                    {editMode ? (
                        <Button id="deleteButton" floated="right" icon="trash" color="red" inverted onClick={handleDeletePressed} />
                    ) : (
                        <Card.Header id="header">
                            <Header as="h2">{qualification.diplomaName}</Header>
                        </Card.Header>
                    )}
                </Card.Content>

                <Card.Description>
                    <List>
                        {editMode && (
                            <List.Item>
                                <CardRow
                                    labelName="diplomaName"
                                    index={index}
                                    data={qualification.diplomaName}
                                    editMode={editMode}
                                    setValue={handleChangeOnQualification}
                                />
                            </List.Item>
                        )}
                        <List.Item>
                            <CardRowForDate
                                labelName="genesisDate"
                                index={index}
                                data={qualification.genesisDate}
                                editMode={editMode}
                                setValue={handleChangeOnQualification}
                            />
                        </List.Item>
                        <List.Item>
                            <CardRow
                                labelName="issuingAuthority"
                                index={index}
                                data={qualification.issuingAuthority}
                                editMode={editMode}
                                setValue={handleChangeOnQualification}
                            />
                        </List.Item>
                    </List>
                </Card.Description>
            </Card.Content>
        </Card>
    );
};

export default QualificationCard;
