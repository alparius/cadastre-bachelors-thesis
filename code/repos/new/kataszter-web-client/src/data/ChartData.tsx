interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
