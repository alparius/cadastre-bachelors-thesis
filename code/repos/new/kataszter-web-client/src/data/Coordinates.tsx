interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
