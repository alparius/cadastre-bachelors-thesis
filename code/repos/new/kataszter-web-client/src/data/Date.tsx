interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
