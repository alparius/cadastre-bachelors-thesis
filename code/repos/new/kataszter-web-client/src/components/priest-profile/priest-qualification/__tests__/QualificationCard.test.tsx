/*===================
  QualificationCard.unit.test.tsx
 ===================*/
import React from "react";
import IQualification, { NewQualification } from "../../../../data/priest/Qualification";
import { NewTestQualification } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntl } from "../../../../util/test/test-wrapper";
import QualificationCard from "../QualificationCard";

const handlers = {
    handleDelete: jest.fn(),
    handleChangeOnQualification: jest.fn()
};

describe("PriestCard element", () => {
    afterEach(() => {
        handlers.handleDelete.mockClear();
        handlers.handleChangeOnQualification.mockClear();
    });
    it("should render a card with the qualification data", () => {
        const qualification: IQualification = NewTestQualification;
        const container = createComponentWithIntlAndRouter(
            <QualificationCard qualification={qualification} index={1} editMode {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render a card with noData label when the qualification is empty", () => {
        const emptyQualification = NewQualification;
        const container = createComponentWithIntlAndRouter(
            <QualificationCard qualification={emptyQualification} index={1} editMode {...handlers} />
        ).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the delete function when delete button pressed", () => {
        const qualification: IQualification = NewTestQualification;
        const container = mountWithIntl(<QualificationCard qualification={qualification} index={1} editMode {...handlers} />);
        container
            .find("#deleteButton")
            .at(0)
            .simulate("click");
        expect(handlers.handleDelete).toHaveBeenCalledTimes(1);
    });

    it.each([
        [0, "2", "3"],
        [1, "2", "3"],
        [2, "2", "3"]
    ])(
        "should  call handleChangeOnQualification on %d. input. After first change: %s, After second change: %s",
        (inputNr, firstChange, secondChange) => {
            const qualification: IQualification = NewTestQualification;
            const container = mountWithIntl(<QualificationCard qualification={qualification} index={1} editMode {...handlers} />);
            const input = container.find("input").at(Number(inputNr));
            input.simulate("change", { target: { value: firstChange } });
            input.simulate("change", { target: { value: secondChange } });
            expect(handlers.handleChangeOnQualification).toHaveBeenCalledTimes(2);
        }
    );
});
