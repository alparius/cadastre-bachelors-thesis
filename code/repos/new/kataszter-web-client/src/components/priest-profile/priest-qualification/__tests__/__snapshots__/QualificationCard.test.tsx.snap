// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestCard element should render a card with noData label when the qualification is empty 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="content"
    >
      <button
        className="ui red icon inverted right floated button"
        id="deleteButton"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="trash icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="description"
    >
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Diploma name
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Genesis date
                :
              </b>
               
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline unstackable fields"
                  style={
                    Object {
                      "marginBottom": "-2px",
                    }
                  }
                >
                  <div
                    className="field"
                  >
                    <b
                      style={Object {}}
                    >
                      year
                    </b>
                    <div
                      className="ui fluid input"
                      style={
                        Object {
                          "width": "6em",
                        }
                      }
                    >
                      <input
                        id=""
                        name="year"
                        onChange={[Function]}
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="field"
                  >
                    <b
                      style={Object {}}
                    >
                      month
                    </b>
                    <div
                      className="ui fluid input"
                      style={
                        Object {
                          "width": "5em",
                        }
                      }
                    >
                      <input
                        id=""
                        name="month"
                        onChange={[Function]}
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    className="field"
                  >
                    <b
                      style={Object {}}
                    >
                      day
                    </b>
                    <div
                      className="ui fluid input"
                      style={
                        Object {
                          "width": "5em",
                        }
                      }
                    >
                      <input
                        id=""
                        name="day"
                        onChange={[Function]}
                        type="number"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="field"
                >
                  <i
                    style={
                      Object {
                        "color": "gray",
                      }
                    }
                  >
                    comment about date (optional)
                  </i>
                  <div
                    className="ui fluid input"
                  >
                    <input
                      id="grey"
                      name="comment"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </form>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Issuing Authority
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`PriestCard element should render a card with the qualification data 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="content"
    >
      <button
        className="ui red icon inverted right floated button"
        id="deleteButton"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="trash icon"
          onClick={[Function]}
        />
      </button>
    </div>
    <div
      className="description"
    >
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Diploma name
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  onChange={[Function]}
                  type="text"
                  value="Erettsegi bizonyitvany"
                />
              </div>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Genesis date
                :
              </b>
               
              <form
                className="ui form"
                onSubmit={[Function]}
              >
                <div
                  className="inline unstackable fields"
                  style={
                    Object {
                      "marginBottom": "-2px",
                    }
                  }
                >
                  <div
                    className="field"
                  >
                    <b
                      style={Object {}}
                    >
                      year
                    </b>
                    <div
                      className="ui fluid input"
                      style={
                        Object {
                          "width": "6em",
                        }
                      }
                    >
                      <input
                        id=""
                        name="year"
                        onChange={[Function]}
                        type="number"
                        value={2000}
                      />
                    </div>
                  </div>
                  <div
                    className="field"
                  >
                    <b
                      style={Object {}}
                    >
                      month
                    </b>
                    <div
                      className="ui fluid input"
                      style={
                        Object {
                          "width": "5em",
                        }
                      }
                    >
                      <input
                        id=""
                        name="month"
                        onChange={[Function]}
                        type="number"
                        value={6}
                      />
                    </div>
                  </div>
                  <div
                    className="field"
                  >
                    <b
                      style={Object {}}
                    >
                      day
                    </b>
                    <div
                      className="ui fluid input"
                      style={
                        Object {
                          "width": "5em",
                        }
                      }
                    >
                      <input
                        id=""
                        name="day"
                        onChange={[Function]}
                        type="number"
                        value={6}
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="field"
                >
                  <i
                    style={
                      Object {
                        "color": "gray",
                      }
                    }
                  >
                    comment about date (optional)
                  </i>
                  <div
                    className="ui fluid input"
                  >
                    <input
                      id="grey"
                      name="comment"
                      onChange={[Function]}
                      type="text"
                      value="2000.06.06"
                    />
                  </div>
                </div>
              </form>
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Issuing Authority
                :
              </b>
               
              <div
                className="ui small fluid input"
              >
                <input
                  onChange={[Function]}
                  type="text"
                  value="Felso Oktatasi Intezmeny"
                />
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
